File: DesignPatterns.BridgePattern/Models/WindowWasher.cs
using DesignPatterns.BridgePattern.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPatterns.BridgePattern.Models
{
    public class WindowWasher : IWasher
    {
        public void SetDirection(int direction)
        {
            Console.WriteLine("Direction: " + direction);
        }

        public void StartWashing()
        {
            Console.WriteLine("Started Washing");
        }

        public void StopWashing()
        {
            Console.WriteLine("Stopped Washing");
        }
    }
}

File: DesignPatterns.UI.Web/ClientApp/src/app/models/Bridge/WindowWasher.ts
import { IWasher } from "./IWasher";

export class WindowWasher implements  IWasher
{
    direction: number = 1;
    position: number = -5;
    minPosition: number = -5;
    maxPosition: number = 105;
    washInterval: any;

    StartWashing(): void {
        var self = this;

        if(this.washInterval != null) this.StopWashing();

        this.washInterval = setInterval(function() {
            if(self.position == self.maxPosition && self.direction == 1) {
                self.StopWashing();
                return;
            }
            if(self.position == self.minPosition && self.direction == -1) {
                self.StopWashing();
                return;
            }

            self.position += self.direction;
        }, 25);
    }
    StopWashing(): void {
        clearInterval(this.washInterval);
    }
    SetDirection(direction: number): void {
        this.direction = direction;
    }
}